contracts/multisig/src/state.rs

```diff
@@ -114,6 +115,48 @@ pub fn save_pub_key(
     Ok(pub_keys().save(store, (signer, pub_key.key_type()), &pub_key.into())?)
 }
 
+pub fn save_prover_to_registry(
```

Comment:
a bit nitpicky, but could we not use `_registry` suffix for these functions and variables? the reason is that I find it a bit confusing because we have a contract called service registry, maybe just call the map `PROVER_CHAIN` and adjust the function names accordingly? what do you think?